### PR TITLE
Query log user guide

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: content/docs/guides/query-log.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+sort_rank: 1
+---
+
+# Using the Prometheus query log
+
+Prometheus has the ability to log all the queries run by the engine to a log
+file. This guide demonstrates how to use that log file, which fields it does
+contain and provides advanced tips about how to operate the log file.
+
+The query log has been introduced in v2.16.0 and is disabled by default.
```
Comment (brian-brazil) [inline content/docs/guides/query-log.md]: This is very awkward. The version should only be mentioned in passing, and the enablement is what the rest is about.

File: content/docs/guides/query-log.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
+---
+title: Query Log
+sort_rank: 1
+---
+
+# Using the Prometheus query log
+
+Prometheus has the ability to log all the queries run by the engine to a log
+file. This guide demonstrates how to use that log file, which fields it does
```
Comment (brian-brazil) [inline content/docs/guides/query-log.md]: it contains

File: content/docs/guides/query-log.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+contain and provides advanced tips about how to operate the log file.
```
Comment (brian-brazil) [inline content/docs/guides/query-log.md]: aren't we doing Oxford commas?

File: content/docs/guides/query-log.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+This example demonstrates how to log all the queries to
+a file called `/var/log/prometheus/query.log`.
+
+First, let's ensure that Prometheus has write access to a directory
+`/var/log/prometheus`. We will assume that Prometheus is running under a user
+called Prometheus:
+
+```shell
+# mkdir /var/log/prometheus
```
Comment (brian-brazil) [inline content/docs/guides/query-log.md]: This fails to mention the need for root. Let's keep to examples that don't require root unless it's strictly required.

File: content/docs/guides/query-log.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+1. Reload the Prometheus server configuration.
+
+### Logging all the queries to a file
+
+This example demonstrates how to log all the queries to
+a file called `/var/log/prometheus/query.log`.
+
+First, let's ensure that Prometheus has write access to a directory
+`/var/log/prometheus`. We will assume that Prometheus is running under a user
+called Prometheus:
```
Comment (brian-brazil) [inline content/docs/guides/query-log.md]: prometheus is not the same as Prometheus.

File: content/docs/guides/query-log.md [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    - targets: ['localhost:9090']
+```
+
+And, reload the Prometheus configuration:
+
+```shell
+$ curl -X POST http://127.0.0.1:9090/-/reload
+```
+
+Or, if Prometheus is not launched with `--web.enable-lifecycle`:
```
Comment (brian-brazil) [inline content/docs/guides/query-log.md]: I'd do this the other way around, mention the flag first.

File: content/docs/guides/query-log.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+```shell
+$ killall -HUP prometheus
```
Comment (brian-brazil) [inline content/docs/guides/query-log.md]: You're making an assumption of OS here, the purpose of killall varies quite a bit.

File: content/docs/guides/query-log.md [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+The first metric, `prometheus_engine_query_log_enabled` is set to 1 of the
+query log is enabled, and 0 otherwise.
+The second one, `prometheus_engine_query_log_failures_total`, indicates the
+number of queries that could not be logged.
+
+## Format of the query log
+
+The query log is a JSON-formatted log. Here is an overview of the fields
+present on any query:
```
Comment (brian-brazil) [inline content/docs/guides/query-log.md]: for a query

File: content/docs/guides/query-log.md [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+        }
+    },
+    "ts": "2020-02-08T14:59:50.387Z"
+}
+```
+
+- params: the query itself. The start and end timestamp, the step and the actual
+  query.
+- stats: statistics. Currently, it contains internal engine timers.
+- ts: timestamp of the log line.
```
Comment (brian-brazil) [inline content/docs/guides/query-log.md]: Capitalisation is inconsistent here.

Surely it's the timestamp of the query, not the log line. Is that when the query starts or ends?

File: content/docs/guides/query-log.md [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+}
+```
+
+
+## Rotating the query log
+
+Prometheus will not rotate the query log itself. Instead, you can use external
+tools to do so.
+
+One of those tools is logrotate. Please refer to the documentation of your
```
Comment (brian-brazil) [inline content/docs/guides/query-log.md]: Link to those docs.

**Comment by roidelapluie** [comment]
I will followup on your comments in another pr

**Pull request merged**